src/useChat.tsx

```diff
@@ -46,6 +46,8 @@ export const useChat = (initValues?: ChatMessage[], initModel?: string) => {
   const [conversationHistories, setConversationHistories] =
     useState<ConversationHistories | null>(null);
 
+  const [isWalletConnectOpen, setIsWalletConnectOpen] = useState(false);
```

Comment:
Moved this state up from App.tsx since we need it available in `useChat` to control the modal from the `useExecuteToolCall`